games/a/pommerman/envs/utility.py

```diff
@@ -317,12 +319,18 @@ def get_next_position(position, direction):
 def make_np_float(feature):
     return np.array(feature).astype(np.float32)
 
-
 def to_json(obj):
-    """Converts the given object into a json representable structure."""
-    try:
-        return json.dumps(obj)
-    except TypeError:
-        # TODO
-        return None
+    return json.dumps(obj, default=serializer)
+
+def serializer(obj):
+    if isinstance(obj, numpy.ndarray):
+        return obj.tolist()
+    elif isinstance(obj, numpy.int64):
+        return int(obj)
+    elif hasattr(obj, 'to_json'):
+        return obj.to_json()
+    elif hasattr(obj, '__dict__'):
```

Comment:
should this be recursive?

Reply:
I believe the default serialization method is called recursively on nested object